python/lsst/pipe/tasks/postprocess.py

```diff
@@ -789,6 +791,181 @@ def _makeArgumentParser(cls):
+
+    This task aggregates the following metadata from all the detectors in a
+    single visit into an exposure catalog:
+    - The visitInfo.
+    - The wcs.
+    - The photoCalib.
+    - The physical_filter and band (if available).
+    - The psf size, shape, and effective area at the center of the detector.
+    - The corners of the bounding box in right ascension/declination.
+    """
+    _DefaultName = "consolidateVisitSummary"
+    ConfigClass = ConsolidateVisitSummaryConfig
+
+    @classmethod
+    def _makeArgumentParser(cls):
+        parser = ArgumentParser(name=cls._DefaultName)
+
+        parser.add_id_argument("--id", "calexp",
+                               help="data ID, e.g. --id visit=12345",
+                               ContainerClass=VisitDataIdContainer)
+        return parser
+
+    def writeMetadata(self, dataRef):
+        """No metadata to persist, so override to remove metadata persistance.
+        """
+        pass
+
+    def writeConfig(self, butler, clobber=False, doBackup=True):
+        """No config to persist, so override to remove config persistance.
+        """
+        pass
+
+    def runDataRef(self, dataRefList):
+        visit = dataRefList[0].dataId['visit']
+
+        self.log.debug("Concatenating metadata from %d per-detector calexps (visit %d)" %
+                       (len(dataRefList), visit))
+
+        expCatalog = self.run(visit, dataRefList, isGen3=False)
+
+        dataRefList[0].put(expCatalog, 'visitSummary', visit=visit)
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        dataRefs = butlerQC.get(inputRefs.calexp)
+        visit = dataRefs[0].dataId.byName()['visit']
+
+        self.log.debug("Concatenating metadata from %d per-detector calexps (visit %d)" %
+                       (len(dataRefs), visit))
+
+        expCatalog = self.run(visit, dataRefs)
+
+        butlerQC.put(expCatalog, outputRefs.visitSummary)
+
+    def run(self, visit, dataRefs, isGen3=True):
+        """Make a combined exposure catalog from a list of dataRefs.
+
+        Parameters
+        ----------
+        visit : `int`
+            Visit identification number
+        dataRefs : `list`
+            List of calexp dataRefs in visit.  May be list of
+            `lsst.daf.persistence.ButlerDataRef` (Gen2) or
+            `lsst.daf.butler.DeferredDatasetHandle` (Gen3).
+        isGen3 : `bool`, optional
+            Specifies if this is a Gen3 list of datarefs.
+
+        Returns
+        -------
+        visitSummary : `lsst.afw.table.ExposureCatalog`
+            Exposure catalog with per-detector summary information.
+        """
+        schema = afwTable.ExposureTable.makeMinimalSchema()
+        schema.addField('visit', type='I', doc='Visit number')
+        schema.addField('detector_id', type='I', doc='Detector number')
+        schema.addField('physical_filter', type='String', size=32, doc='Physical filter')
+        schema.addField('band', type='String', size=32, doc='Name of band')
```

Comment:
I wish we could have better docstrings for filter and band here, but I'm not sure what they would be.

Reply:
I don't know what to say, except that this will have to be revisited once the Great Filter Migration is done, and hopefully all I have to update then are the doc strings.